src/store/modules/ProfilesModule.ts

```diff
@@ -0,0 +1,51 @@
+import FileUtils from "../../utils/FileUtils";
+import R2Error from "../../model/errors/R2Error";
+import { ActionTree } from "vuex";
+import { State as RootState } from "../../store";
+import Profile from "../../model/Profile";
+import FsProvider from "../../providers/generic/file/FsProvider";
+
+interface State {
+    profileList: string[];
+}
+
+/**
+ * State for Profiles, i.e. list for profiles in a single game/community.
+ */
+export const ProfilesModule = {
+    namespaced: true,
+
+    state: (): State => ({
+        profileList: ['Default'],
+    }),
+    mutations: {
+        setProfileList(state: State, profileList: string[]) {
+            state.profileList = profileList;
+        },
+    },
+    actions: <ActionTree<State, RootState>>{
+        async removeSelectedProfile({rootGetters, state, dispatch}) {
+            const activeProfile: Profile = rootGetters['profile/activeProfile'];
+            const path = activeProfile.getPathOfProfile();
+            const profileName = activeProfile.getProfileName();
+
+            try {
+                await FileUtils.emptyDirectory(path);
+                await FsProvider.instance.rmdir(path);
+            } catch (e) {
+                throw R2Error.fromThrownValue(e, 'Error whilst deleting profile from disk');
+            }
+
+            if (profileName.toLowerCase() !== 'default') {
```

Comment:
When `Default` profile is "deleted", new empty `Default` profile should be created instantly - and it does. However, when that profile is selected, it still shows the mods of the `Default` profile that was deleted. I think this if-clause might be the culprit. Either we should always call `setSelectedProfile` at the end of `removeSelectedProfile`; or we could add an else-clause here that calls just `profile/updateModListFromFile` and `tsMods/prewarmCache`.

Reply:
You went a little overboard with this. Now, in addition to always calling `setSelectedProfile`, you also always filter the removed profile name from the list of profiles. We *don't* want to do that if it was the default profile - the `Default` should always be visible on the list.

(IDK if there's a good rationale to treat the default profile differently, but at least the code base assumes that there's always a profile called Default, so it's nice for UI to reflect that. And that's how it has worked so far, so maybe we shouldn't change that in this PR.)

Reply:
should be ok now